src/main/core/main.rs

```diff
@@ -0,0 +1,258 @@
+use std::ffi::{CStr, OsStr};
+use std::os::unix::ffi::OsStrExt;
+
+use anyhow::{self, Context};
+use clap::Clap;
+use nix::sys::{personality, signal};
+
+use crate::core::logger::log_wrapper::c_to_rust_log_level;
+use crate::core::logger::shadow_logger;
+use crate::core::support::configuration::{CliOptions, ConfigFileOptions, ConfigOptions};
+use crate::cshadow as c;
+
+/// Main entry point for the simulator.
+pub fn run_shadow<'a, I>(args: I) -> anyhow::Result<()>
```

Comment:
Nitpick, but for code that's called from one place and never going to be on the hot path, perhaps better to simplify to just taking a concrete type here instead of making generic over iterators

Reply:
I made this generic so that we can easily drop in `std::env::args_os()` (which implements `Iter`) later when the real main function is eventually implemented in rust. But we can make this take a vector instead for now.